app/uk/gov/hmrc/eoricommoncomponent/frontend/connector/EmailVerificationConnector.scala

```diff
@@ -17,86 +17,100 @@
 package uk.gov.hmrc.eoricommoncomponent.frontend.connector
 
 import javax.inject.{Inject, Singleton}
-import play.api.Logger
-import play.api.libs.json.{JsObject, Json}
+import play.api.libs.json.Json
 import uk.gov.hmrc.eoricommoncomponent.frontend.config.AppConfig
-import uk.gov.hmrc.eoricommoncomponent.frontend.connector.EmailVerificationKeys.{
-  LinkExpiryDurationKey,
-  TemplateIdKey,
-  _
+import uk.gov.hmrc.eoricommoncomponent.frontend.models.email.{
+  ResponseWithURI,
+  StartVerificationJourneyEmail,
+  StartVerificationJourneyRequest,
+  VerificationStatusResponse
 }
-import uk.gov.hmrc.eoricommoncomponent.frontend.connector.httpparsers.EmailVerificationRequestHttpParser.EmailVerificationRequestResponse
-import uk.gov.hmrc.eoricommoncomponent.frontend.connector.httpparsers.EmailVerificationStateHttpParser.EmailVerificationStateResponse
-import uk.gov.hmrc.http.HeaderCarrier
-import uk.gov.hmrc.http.HttpClient
+import uk.gov.hmrc.http.{HeaderCarrier, StringContextOps}
+import uk.gov.hmrc.http.client.HttpClientV2
+import java.net.URL
+import cats.data.EitherT
```

Comment:
I do know what you mean, the ideal would be to keep it consistent but I feel like this may be positive because at least it starts setting a new standard of using eithers to handle error scenarios rather than throwing exceptions - though this could be done just using Future[Either[A, B]]. We can have a chat tomorrow about it